File: app/src/main/java/com/github/overpass/gather/model/usecase/permission/LocationPermissionUseCase.java
package com.github.overpass.gather.model.usecase.permission;

import com.github.overpass.gather.map.PermissionRequestResult;
import com.github.overpass.gather.model.commons.SingleLiveEvent;

public class LocationPermissionUseCase {

    private final SingleLiveEvent<PermissionRequestResult> permissionRequestResultData;

    public LocationPermissionUseCase() {
        this.permissionRequestResultData = new SingleLiveEvent<>();
    }

    public SingleLiveEvent<PermissionRequestResult> getPermissionRequestResultData() {
        return permissionRequestResultData;
    }

    public void onRequestPermissionsResult(int requestCode,
                                    String[] permissions,
                                    int[] grantResults) {
        permissionRequestResultData.setValue(new PermissionRequestResult(requestCode,
                permissions, grantResults));
    }
}

File: app/src/main/java/com/github/overpass/gather/model/usecase/permission/LocationPermissionUseCase.kt
package com.github.overpass.gather.model.usecase.permission

import com.github.overpass.gather.commons.android.lifecycle.SingleLiveEvent
import com.github.overpass.gather.ui.map.PermissionRequestResult

class LocationPermissionUseCase(
        val permissionRequestResultData: SingleLiveEvent<PermissionRequestResult>
) {

    fun onRequestPermissionsResult(requestCode: Int,
                                   permissions: Array<String>,
                                   grantResults: IntArray) {
        permissionRequestResultData.value = PermissionRequestResult(requestCode,
                permissions, grantResults)
    }
}
